pkg/ansible/runner.go

```diff
@@ -0,0 +1,201 @@
+package ansible
+
+import (
+	"path"
+
+	log "github.com/sirupsen/logrus"
+
+	kbatch "k8s.io/api/batch/v1"
+	kapi "k8s.io/api/core/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	openshiftAnsibleImage          = "openshift/origin-ansible:v3.7"
+	openshiftAnsibleServiceAccount = "cluster-operator-controller-manager"
+	inventoryConfigMap             = "ansible-inventory"
+	sshPrivateKeySecret            = "ssh-private-key"
+	operatorNamespace              = "cluster-operator"
+	openshiftAnsibleContainerDir   = "/usr/share/ansible/openshift-ansible/"
+	jobNamePrefix                  = "openshift-ansible-"
+	awsCredentialsSecret           = "aws-credentials"
+)
+
+type ansibleRunner struct {
+	KubeClient kubernetes.Interface
+	Image      string
+}
+
+func NewAnsibleRunner(kubeClient kubernetes.Interface) *ansibleRunner {
+	return &ansibleRunner{
+		KubeClient: kubeClient,
+		Image:      openshiftAnsibleImage,
+	}
+}
+func (r *ansibleRunner) createInventoryConfigMap(inventory string, vars string) (*kapi.ConfigMap, error) {
+	log.Debugln("Creating inventory/vars ConfigMap")
+	cfgmap := &kapi.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: inventoryConfigMap,
```

Comment:
Does each cluster need its own configmap? We can't use a constant name for the configmap, can we?

Reply:
Absolutely not, I think each execution of an ansible job will need it's own. Haven't gotten this far yet, but my plan was to pre-create the config map (obviously), label it with something indicating it's for a particular job (maybe pre-generate random job names), and then add a controller that would watch configmaps with that label, and cleanup any whose job no longer exists. (open to better ideas if anyone has any)

Reply:
You can simply switch to a GenerateName and use the constant as a prefix. It may also be useful to add a label to identify what cluster this configmap belongs to so that you can identify cluster-related resources later.

Reply:
Agree with that. I would expect this name to be something like "someconstant-nameofcluster-somerandom".